os/hal/ports/STM32/LLD/MACv2/hal_mac_lld.c

```diff
@@ -651,10 +651,12 @@ void mac_lld_release_receive_descriptor(MACReceiveDescriptor *rdp) {
  * @notapi
  */
 bool mac_lld_poll_link_status(MACDriver *macp) {
-  uint32_t maccr, bmsr, bmcr;
+  uint32_t maccr;
 
   maccr = ETH->MACCR;
-
+ /* Fixed link connection defined in board.h.*/
+#if !defined(BOARD_PHY_FIXED_LINK)
```

Comment:
So I think what you need to do is add a config item to hal_mmc_lld.h, something like:
```
/**
 * @brief   <docs in here>
 */
#if !defined(STM32_MAC_PHY_FIXED_LINK) || defined(__DOXYGEN__)
#define STM32_MAC_PHY_FIXED_LINK FALSE
#endif
/**
 * @brief   <docs in here>
 */
#if !defined(STM32_MAC_PHY_FIXED_LINK_TYPE) || defined(__DOXYGEN__)
#define STM32_MAC_PHY_FIXED_LINK_TYPE LINK_100_FULLDUPLEX
#endif
```
and then adjust the source appropriately for these

Reply:
changes made accordingly

Reply:
Thanks - I will ping the ChibiOS guys

Reply:
Thanks !